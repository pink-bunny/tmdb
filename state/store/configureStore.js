import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';

import reducer from './reducer';

const loggerMiddleware = (store) => (next) => (action) => {
  /* eslint-disable no-console */
  console.group(action.type);
  console.info('dispatching', action);
  const result = next(action);
  console.log('next state', store.getState());
  console.groupEnd();
  /* eslint-enable */

  return result;
};

export default function configureStore() {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(reducer, undefined, composeWithDevTools(middlewareEnhancer));

  return store;
}
