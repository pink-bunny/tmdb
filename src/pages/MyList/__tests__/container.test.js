import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import ConnectedMyList from '../container';
import {
  fetchMyList,
  removeFromList,
  clearListIdFromState
} from '../../../../state/my_list/actions';

jest.mock('../../../../state/my_list/actions', () => ({
  fetchMyList: jest.fn(),
  removeFromList: jest.fn(),
  clearListIdFromState: jest.fn()
}));
jest.mock('../../../../state/my_list/selectors', () => ({
  isMyListLoading: jest.fn(() => false),
  myListError: jest.fn(() => ''),
  myListInfo: jest.fn(() => ({ id: 1, name: 'Custom list' })),
  myListMovies: jest.fn(() => [{ id: 1, title: 'Mock title' }])
}));

describe('MyList container', () => {
  const listId = '1';
  const store = configureStore()({});
  const props = {
    store,
    match: { params: { listId } }
  };
  store.dispatch = jest.fn();
  const wrapper = shallow(<ConnectedMyList {...props} />); /* eslint-disable-line react/jsx-props-no-spreading, max-len */
  const container = wrapper.dive().dive();
  const instance = container.instance();

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('handleFetchList() called fetchMyList() with list id', () => {
    instance.handleFetchList();
    expect(fetchMyList).toHaveBeenCalledWith(listId);
  });

  it('handleRemoveFromList() called removeFromList() with right arguments', () => {
    const movieId = 1;
    instance.handleRemoveFromList(movieId);
    expect(removeFromList).toHaveBeenCalledWith(listId, movieId);
  });

  it('componentWillUnmount() called clearListIdFromState()', () => {
    instance.componentWillUnmount();
    expect(clearListIdFromState).toHaveBeenCalled();
  });
});
