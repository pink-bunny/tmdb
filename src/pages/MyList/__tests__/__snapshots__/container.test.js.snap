// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList container matches snapshot 1`] = `
<MyList
  currentPage={0}
  error=""
  fetchList={[Function]}
  listInfo={
    Object {
      "id": 1,
      "name": "Custom list",
    }
  }
  listMovies={
    Array [
      Object {
        "id": 1,
        "title": "Mock title",
      },
    ]
  }
  loading={false}
  removeModalAction={[Function]}
  totalItems={0}
/>
`;
